judo.go

```diff
@@ -6,6 +6,7 @@ import (
 
 	"github.com/amagimedia/judo/v2/client"
 	judoMsg "github.com/amagimedia/judo/v2/message"
+	primaryBackupPub "github.com/amagimedia/judo/v2/protocols/pub/primarybackuppub"
```

Comment:
Some changes here 
1. Let the PrimaryBackupSub and PrimaryBackupPub be implementations of JudoClient and JudoPub
2. Let us name it AmagiPublisher and AmagiSubscriber
3. These will have a case statement switch case "amagi" method sub or pub
4. Inside these implement the same function Start() / Configure() OnMessage() etc. 
5. In Configure you take a config which has primary and backup 
6. In OnMessage you check the isDuplicate and call the appropriate primary or backup OnMessage or else drop the message